gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/util/ImportGradesHelper.java

```diff
@@ -158,14 +161,14 @@ private static ImportedSpreadsheetWrapper parseCsv(final InputStream is, final G
 		// manually parse method so we can support arbitrary columns
 		CSVReader reader;
 		if(StringUtils.isEmpty(userDecimalSeparator)){
-			reader = new CSVReader(new InputStreamReader(is));
+			reader = new CSVReader(new InputStreamReader(is, "ISO-8859-1"));
```

Comment:
What happens when the file is UTF-8?

Reply:
See https://openjdk.java.net/jeps/8187041

Reply:
For those thinking well UTF-8 is a single character, but UTF-8 is part of a variable length and so it works well with other encodings where single byte encodings would fail.

Reply:
As mentioned in the task associated with the branch: 

"When a user exports a ratings CSV file, modifies it afterwards and imports the file back again with the changes, if any text contains tilde or special characters, Gradebook duplicates the columns containing those special characters, just replacing the special characters with strangely parsed characters. This is so because, when saving any local changes on the exported CSV file, Microsoft Excel (the most widely used spreadsheet software) changes the document's coding from UTF-8 with BOM, it's original coding, to ANSI. Because of that, when processing the file, it is processed by default in UTF-8, thus, transcoding incorrectly those special characters."

To reproduce this problem:
1º Export a CSV with the gradebooks (Check that the coding is UTF8-BOM).
2º Change a gradebook of the downloaded CSV with the MSEXCEL application and give it to close so that it gives us the option to save. We save again the file in CSV (We see that the codification has changed to ANSI).
3º When we try to import the file, it duplicates the subjects that contain punctuation symbols with the strange symbols.

Reply:
We know UTF-8 is optimal for this problem but we can't make Microsoft understand that so, sadly, we need to use ISO-8859-1.